src/main/java/com/yelp/nrtsearch/server/utils/VersionedResourceObject.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.utils;
+
+import java.util.Date;
+
+public class VersionedResourceObject {
+
+  private final String serviceName;
+  private final String resourceName;
+  private final String versionHash;
+  private final Date creationTimestamp;
+
+  public VersionedResourceObject(
```

Comment:
Can implement `toString` method at least, if not `equals` and `hashcode` as well. These can be autogenerated by IDEs.